### DISCUSSION TITLE
Is there any example for passing a map array to a cuda kernel?

### DISCUSSION THREAD

**Discussion by wangzilu** [NVIDIA/cuCollections#493] [Q&A] [answered]
I have several hash maps to pass them into kernel. Is there any example to show how can I combine them into a vector or a array and them pass it into cuda kernel.

**Comment by sleeepyjack** [comment, marked as answer]
Hi @wangzilu,

There are multiple ways to achieve this. I just put together a simple example [here](https://godbolt.org/z/7ae86j6sd).
The idea is to emplace the map objects into a host vector and then create non-owning reference objects for each map that can be placed into a device vector and then later be used in the kernel.

The downside of this approach is that each map will allocate its own storage, which can lead to overheads during allocation/initialization (plain `cudaMalloc` is a blocking operation).
Instead, it is also possible to allocate a single storage array first and then subdivide it into multiple maps.
This approach is a bit more complex. We have an example [here](https://github.com/NVIDIA/cuCollections/blob/dev/examples/static_set/device_subsets_example.cu).

**Reply by PointKernel** [reply]
@wangzilu marking this question as answered. Feel free to let us know if you have any more questions.
